### ISSUE TITLE
Neovide fails to build on `aarch64-darwin`

### ISSUE THREAD

**Issue by shaunsingh** [NixOS/nixpkgs#145523]
### Describe the bug
Neovide fails to build on aarch64-darwin

### Steps To Reproduce
1. `nix-shell -p neovide --run neovide`

 >   = note: ld: framework not found Security
       >           clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
       >
       >
       > error: could not compile `skia-bindings` due to previous error
       > warning: build failed, waiting for other jobs to finish...
       > error: build failed
       For full logs, run 'nix log /nix/store/ji5zkfwb0h3nyzi16zibvy084mimygfc-neovide-unstable-2021-10-09.drv'.
       
Full output: 
https://pastebin.com/JVBdqrYS

### Expected behavior
It should build as expected

### Additional context
Locally, neovide builds from source on aarch64-darwin just fine

### Notify maintainers

@ck3d 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"aarch64-darwin"`
 - host os: `Darwin 21.1.0, macOS 12.0.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.5pre20211007_844dd90`
 - channels(shauryasingh): `"darwin"`
 - channels(root): `"nixpkgs-21.11pre327990.4789953e5c1"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by Artturin** [comment]
please test https://github.com/NixOS/nixpkgs/pull/145526

**Comment by shaunsingh** [comment]
Sorry, I'm still quite new to nix 
How do I build/test a PR?

**Comment by Artturin** [comment]
https://github.com/Mic92/nixpkgs-review#usage

**Comment by shaunsingh** [comment]
Thanks, that error is fixed but there is another one: 
https://pastebin.com/Gi3PY9KM
```bash
  Python 2 found: "python"
  Skia args: is_official_build=true is_debug=false skia_enable_gpu=true skia_use_gl=true skia_use_egl=false skia_use_x11=false skia_use_system_libpng=false skia_use_libwebp_encode=false skia_use_libwebp_decode=false skia_use_system_zlib=false skia_use_xps=false skia_use_dng_sdk=false cc="clang" cxx="clang++" skia_use_icu=false skia_use_system_libjpeg_turbo=false target_os="mac" target_cpu="arm64" skia_use_expat=true skia_use_system_expat=false extra_cflags=["-O3"]
  ERROR at //gn/skia/BUILD.gn:44:7: Script returned non-zero exit code.
        exec_script("../find_xcode_sysroot.py", [ "macosx" ], "trim string")
        ^----------
  Current dir: /private/tmp/nix-build-neovide-unstable-2021-10-09.drv-0/source/target/aarch64-apple-darwin/release/build/skia-bindings-9a9b21bf4df377c3/out/skia/
  Command: python /nix/store/kw1vcfbfz37w2sgk6scigkvmc10hadxl-source/gn/find_xcode_sysroot.py macosx
  Returned 1.
  stderr:

  Traceback (most recent call last):
    File "/nix/store/kw1vcfbfz37w2sgk6scigkvmc10hadxl-source/gn/find_xcode_sysroot.py", line 15, in <module>
      print(subprocess.check_output(['xcrun', '--sdk', sdk, '--show-sdk-path']))
    File "/nix/store/97d463pna02mfv6zbaskm5xsivkhc58c-python-2.7.18/lib/python2.7/subprocess.py", line 216, in check_output
      process = Popen(stdout=PIPE, *popenargs, **kwargs)
    File "/nix/store/97d463pna02mfv6zbaskm5xsivkhc58c-python-2.7.18/lib/python2.7/subprocess.py", line 394, in __init__
      errread, errwrite)
    File "/nix/store/97d463pna02mfv6zbaskm5xsivkhc58c-python-2.7.18/lib/python2.7/subprocess.py", line 1047, in _execute_child
      raise child_exception
  OSError: [Errno 2] No such file or directory

  See //gn/BUILDCONFIG.gn:175:3: which caused the file to be included.
    "//gn/skia:default",
    ^------------------

  --- stderr
  thread 'main' panicked at '""', /private/tmp/nix-build-neovide-unstable-2021-10-09.drv-0/neovide-unstable-2021-10-09-vendor.tar.gz/skia-bindings/build_support/skia/config.rs:369:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
error: builder for '/nix/store/6lfz6b7v88s2i1l8ni5nqyr2s7xv85r8-neovide-unstable-2021-10-09.drv' failed with exit code 101;
       last 10 log lines:
       >
       >   See //gn/BUILDCONFIG.gn:175:3: which caused the file to be included.
       >     "//gn/skia:default",
       >     ^------------------
       >
       >   --- stderr
       >   thread 'main' panicked at '""', /private/tmp/nix-build-neovide-unstable-2021-10-09.drv-0/neovide-unstable-2021-10-09-vendor.tar.gz/skia-bindings/build_support/skia/config.rs:369:9
       >   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
       > warning: build failed, waiting for other jobs to finish...
       > error: build failed
       For full logs, run 'nix log /nix/store/6lfz6b7v88s2i1l8ni5nqyr2s7xv85r8-neovide-unstable-2021-10-09.drv'.

```

**Comment by Artturin** [comment]
maybe submodules need to be cloned, not sure. leaving this to someone from the @NixOS/darwin-maintainers team

**Comment by prusnak** [comment]
The build is looking for `xcrun` which is not present. Usually adding `xcbuild` to `buildInputs` does the trick, but doing simply that did not help in this particular case.

**Comment by uri-canva** [comment]
Something is quite strange about this package, I've been trying to fix the build, but it has a lot of non deterministic failures. Sometimes it can't find this or that header which I've verified existed when the build started, sometimes it can't find some object output. It's like something is deleting the files while the build is running. I have not had this with other packages, and I'm having these same issues on both my Intel and M1 machines.

**Comment by ck3d** [comment]
Fixed by #191561
